sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClient.java

```diff
@@ -1249,25 +1242,45 @@ private boolean isValidKeyLocallyAvailable() {
         }
 
         if (key == null && implClient.getKeyCollection() != null) {
-            key = CryptographyUtils.SECRETS_COLLECTION.equals(implClient.getKeyCollection())
-                ? getSecretKey() : getKey().getKey();
+            try {
+                // Get key from service and validate it. Then attempt to create a local cryptography client or default
+                // to service.
+                key = CryptographyUtils.SECRETS_COLLECTION.equals(implClient.getKeyCollection())
+                    ? implClient.getSecretKey() : getKey().getKey();
+            } catch (HttpResponseException e) {
+                if (e.getResponse().getStatusCode() == 403) {
+                    localOperationNotSupported = true;
+                    
+                    LOGGER.warning("Could not retrieve key from service to perform cryptographic operations locally. "
```

Comment:
Should this be a warning or just info?

Reply:
I was thinking some users might expect the operations to be performed locally by default for the sake of performance and well, letting them know it's not the case seemed worthy of using the WARN level. In my mind, we use WARN when something undesirable happens but we are still able to continue. What do you think?

Reply:
I think this is pointing more towards INFO

https://azure.github.io/azure-sdk/java_implementation.html#java-logging-warn

Another option, is only log once for each CryptographyClient if this happens.

Reply:
After reading the docs, I also think INFO is appropriate. We do log things once per client if the key cannot be retrieved though :)